### ISSUE TITLE
Material Drawer : VoiceOver : Should be able to announce 'Dismiss' at the end of drawer contents and set accessibility focus on the drawer

### ISSUE THREAD

**Issue by sharvari-g** [flutter/flutter#78863]
## Use case

While using voice over, when the user opens a [material drawer](https://api.flutter.dev/flutter/material/Drawer-class.html), voice over announces 'Dismiss' button first without setting accessibility focus on any element, and then moves on to announce other actionable items from the drawer. This feature request is for requesting slight modifications in this behaviour and would be applicable to all use cases involving the drawer.

## Proposal

Would it be possible to support following features for the drawer -
1. After opening the drawer, the voice over should announce the first element of the drawer. The 'Dismiss' action should be announced at the end. This is also in accordance with [WAI-ARAI practices](https://www.w3.org/TR/wai-aria-practices/#menu) 
   `When a menu opens, or when a menubar receives focus, keyboard focus is placed on the first item. All items are focusable 
as described in § 6.6 Keyboard Navigation Inside Components.
`
2. While the voice over announces 'Dismiss' action, the focus should be set on the entire drawer.

**Comment by chunhtai** [comment]
we need to use semantics sort key similar to modalRoute barrier to set the correct focus order
